Apps/W1/Shopify/app/src/Staff/Tables/ShpfyStaffMember.Table.al

```diff
@@ -0,0 +1,135 @@
+        field(5; "Email"; Text[100])
+        {
+            Caption = 'Email';
+            ToolTip = 'Specifies the staff member''s email address.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(6; Exists; Boolean)
+        {
+            Caption = 'Exists';
+            ToolTip = 'Indicates if the staff member exists.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(7; "First Name"; Text[100])
+        {
+            Caption = 'First Name';
+            ToolTip = 'Specifies the staff member''s first name.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(8; Initials; Text[10])
+        {
+            Caption = 'Initials';
+            ToolTip = 'Specifies the staff member''s initials.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(9; "Shop Owner"; Boolean)
+        {
+            Caption = 'Shop Owner';
+            ToolTip = 'Indicates if the staff member is the shop owner.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(10; "Last Name"; Text[100])
+        {
+            Caption = 'Last Name';
+            ToolTip = 'Specifies the staff member''s last name.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(11; Locale; Text[5])
+        {
+            Caption = 'Locale';
+            ToolTip = 'Specifies the staff member''s locale.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(12; Name; Text[100])
+        {
+            Caption = 'Name';
+            ToolTip = 'Specifies the staff member''s name.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(13; Phone; Text[20])
+        {
+            Caption = 'Phone';
+            ToolTip = 'Specifies the staff member''s phone number.';
+            DataClassification = CustomerContent;
+            Editable = false;
+        }
+        field(14; "Salesperson Code"; Code[20])
+        {
+            Caption = 'Salesperson Code';
+            ToolTip = 'Specifies the sales person or purchaser code.';
+            DataClassification = CustomerContent;
+            TableRelation = "Salesperson/Purchaser" where(Blocked = const(false));
+
+            trigger OnValidate()
+            var
+                SalespersonPurchaser: Record "Salesperson/Purchaser";
+                ShpfyStaffMember: Record "Shpfy Staff Member";
+                SalespersonPurchaserMappingErr: Label '%1 = %2 already mapped for the Shopify Staff Member %3.', Comment = '%1 = Salesperson/Purchaser table caption, %2 = Salesperson/Purchaser code, %3 = Shopify Staff Member name';
+            begin
+                if "Salesperson Code" <> '' then begin
```

Comment:
On the SalesHeader there's an additional check when you validate a salesperson.
https://github.com/StefanMaron/MSDyn365BC.Code.History/blob/87e7d2a19d2097560d47116202f47d4a20940a14/BaseApp/Source/Base%20Application/Sales/Document/SalesHeader.Table.al#L8468

We could discuss if it's useful to have this check already here, too.

Otherwise someone could specify a salesperson here that cannot be used in a sales document.